fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -36,13 +43,26 @@ import org.mozilla.fenix.theme.Theme
  * Top-level UI for displaying the navigation bar.
  *
  * @param actionItems A list of [ActionItem] used to populate the bar.
- * @param menuButton A [MenuButton] to be used for [ItemType.MENU]
+ * @param menuButton A [MenuButton] to be used for [ItemType.MENU].
+ * @param browserStore A store to observe the number of opened tabs.
+ * @param browsingModeManager A helper class that provides access to the current [BrowsingMode].
  */
 @Composable
 fun NavigationBar(
     actionItems: List<ActionItem>,
     menuButton: MenuButton? = null,
+    browserStore: BrowserStore,
+    browsingModeManager: BrowsingModeManager,
```

Comment:
I might argue for this to be simplified:
```suggestion
    tabCount: Int,
```

As a leaf node in the Composable tree, `NavigationBar` doesn't need to have knowledge of the store here, and the mode is simply used as a logic control with the observation.

This will help simplify the previews, too.

Reply:
I was thinking about the same thing, but I couldn't come up with a good way to organize the hierarchy. Where should we observe the store? Right in the `BottomToolbarContainerView`? I couldn't find any examples of observing things outside of some form of composable, that's hence I put it in the first composable we have.

Reply:
Yeah, in `BottomToolbarContainerView` makes sense to me. It would likely live there or further up, if that View were to ever be replaced with a Composable down the road.

There's a few examples of this with our ViewHolders:
https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabViewHolder.kt#45-61